src/python/pants/backend/python/tasks/python_tool_prep_base.py

```diff
@@ -99,14 +101,20 @@ def _build_tool_pex(self, tool_subsystem, interpreter, pex_path):
 
   def execute(self):
     tool_subsystem = self.tool_subsystem_cls.scoped_instance(self)
-    pex_name = tool_subsystem.options_scope
-    pex_path = os.path.join(self.workdir, self.fingerprint, '{}.pex'.format(pex_name))
+    specs_fingerprint = stable_json_sha1(tool_subsystem.get_requirement_specs())
+    pex_name_base = tool_subsystem.options_scope
+    full_pex_filename = '{}-{}.pex'.format(pex_name_base, specs_fingerprint)
+    pex_path = os.path.join(
+      get_pants_cachedir(),
+      'python-tools',
```

Comment:
We already have `tools/jvm` under `~/.cache/pants`. How about `tools/python` for a little less entropy.

Reply:
I will make this change!